interface/src/app/tribe_client/tribe_client.js

```diff
@@ -0,0 +1,39 @@
+angular.module('adage.tribe_client', [
+    'adage.tribe_client.resource',
+    'adage.tribe_client.directives',
+])
+
+.factory( 'UserFactory', ['User', function( User ) {
+    var promise = false;
+    var user = null;
+
+    return {
+        getUser: function() { //only call this when the promise completes
+            return user;
+        },
+        getPromise: function() {
+            if (!promise) {
+                promise = User.get({}, function( data ) {
+                    if (data.meta.total_count !== 0) {
+                        user = data.objects[0];
+                    }
+                });
+            }
+            return promise;
+        },
+        resetPromise: function() { // reset the promise in case we need to check the user again
+            promise = false;
```

Comment:
Well, my impression was that null does not evaluate to false in JavaScript, right? This is basically just to see if the promise has been returned, which should be a true/false value. On the other hand, the 'user' variable actually holds the user object, and that's why that one is set to 'null' if no User has been returned. Do you think that approach makes sense?